evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
-        return vm_class
+        for n in reversed(cls.vms_by_range.keys()):
```

Comment:
Can you change this to use a binary search.  The `bisect` module from the standard library will do this for you.

https://docs.python.org/3.0/library/bisect.html

Assuming the ranges are sorted which can be done as a normalization step during configuration, this should be trivial to implement.